src/platform/MpiServer.c

```diff
@@ -0,0 +1,611 @@
+        OsConfigLogError(GetPlatformLog(), "Invalid null request object");
+        status = HTTP_BAD_REQUEST;
+    }
+    else if (NULL == (clientName = JsonGetString(requestObject, g_clientName)))
+    {
+        OsConfigLogError(GetPlatformLog(), "Failed to parse client name from request body");
+        status = HTTP_BAD_REQUEST;
+    }
+    else if (0 > (maxPayloadSizeBytes = JsonGetNumber(requestObject, g_maxPayloadSizeBytes)))
+    {
+        OsConfigLogError(GetPlatformLog(), "Failed to parse max payload size from request body");
+        status = HTTP_BAD_REQUEST;
+    }
+    else
+    {
+        if (IsFullLoggingEnabled())
+        {
+            OsConfigLogInfo(GetPlatformLog(), "Received MpiOpen request for client '%s' with max payload size %d", clientName, maxPayloadSizeBytes);
+        }
+
+        if (NULL != (uuid = (char*)mpiOpen(clientName, maxPayloadSizeBytes)))
+        {
+            if (IsFullLoggingEnabled())
+            {
+                OsConfigLogInfo(GetPlatformLog(), "Created session '%s' for client '%s'", uuid, clientName);
+            }
+
+            estimatedSize = strlen(responseFormat) + strlen(uuid) + 1;
+
+            if (NULL != (*response = (char*)malloc(estimatedSize)))
+            {
+                snprintf(*response, estimatedSize, responseFormat, uuid);
+                *responseSize = strlen(*response);
+            }
+            else
+            {
+                OsConfigLogError(GetPlatformLog(), "Failed to allocate memory for response");
+                status = HTTP_INTERNAL_SERVER_ERROR;
+                *responseSize = 0;
+            }
+
+            FREE_MEMORY(uuid);
+        }
+        else
+        {
+            OsConfigLogError(GetPlatformLog(), "MpiOpen failed to create a session for client '%s'", clientName);
+            status = HTTP_INTERNAL_SERVER_ERROR;
+        }
+    }
+
+    return status;
+}
+
+static HTTP_STATUS MpiCloseRequest(JSON_Object* requestObject, char** response, int* responseSize, void(*mpiClose)(MPI_HANDLE))
+{
+    HTTP_STATUS status = HTTP_OK;
+    const char* clientSession = NULL;
+
+    UNUSED(response);
+    UNUSED(responseSize);
+
+    if (NULL == requestObject)
+    {
+        OsConfigLogError(GetPlatformLog(), "Invalid null request object");
+        status = HTTP_BAD_REQUEST;
+    }
+    else if (NULL == (clientSession = JsonGetString(requestObject, g_clientSession)))
+    {
+        OsConfigLogError(GetPlatformLog(), "Failed to parse client session from request body");
+        status = HTTP_BAD_REQUEST;
+    }
+    else
+    {
+        if (IsFullLoggingEnabled())
+        {
+            OsConfigLogInfo(GetPlatformLog(), "Received MpiClose request, session '%s'", clientSession);
+        }
+
+        mpiClose((MPI_HANDLE)clientSession);
+    }
+
+    return status;
+}
+
+static HTTP_STATUS MpiSetRequest(JSON_Object* requestObject, char** response, int* responseSize, int(*mpiSet)(MPI_HANDLE, const char*, const char*, MPI_JSON_STRING, const int))
```

Comment:
>int(*mpiSet)(MPI_HANDLE, const char*, const char*, MPI_JSON_STRING, const int) [](http://example.com/codeflow?start=97&length=78)

Don't do this. Instead typedef a function pointer type and use it here. Do the same for all